runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -785,6 +789,13 @@ J9::CodeGenerator::preLowerTrees()
    _uncommonedNodes.init(64, true);
    }
 
+void
+J9::CodeGenerator::postLowerTrees()
+   {
+   OMR::CodeGeneratorConnector::postLowerTrees();
+   if ((self()->isRerunLivenessAnalysis() && (self()->comp()->getOptLevel() > noOpt))
+      self()->rerunLivenessAnalysis();
+   }
```

Comment:
@andrewcraik @liqunl This change sets a flag in CG to trigger rerunLivenessAnalysis whenever splitPostGRA is called. However based on my test, I don't think liveness analysis should be run once the flag is true. There might be other things to take into consideration, such as if the optLevel is noOpt, it doesn't seem to be right to rerun liveness analysis. 

My question is what criteria should be used to allow liveness analysis to rerun after tree lowering? 

I found there are two places in our code that do liveness analysis. Should we rerun liveness analysis after tree lowering if the liveness analysis has been performed in both  TR_GlobalRegisterAllocator and TR_GlobalLiveVariablesForGC? Or they are irrelevant to rerun liveness analysis after tree lowering? 

1. TR_GlobalRegisterAllocator
https://github.com/eclipse/openj9-omr/blob/ff84a120148d2eeba5a6fcc594b846cb93f809c5/compiler/optimizer/GlobalRegisterAllocator.cpp#L446

2. TR_GlobalLiveVariablesForGC https://github.com/eclipse/openj9/blob/33d30b17a7c6329c55546a46c42e5fdae0ac8585/runtime/compiler/optimizer/LiveVariablesForGC.cpp#L266

@Leonardo2718 @hzongaro FYI

Reply:
If live info has been computed, and then gets invalidated by calling splitPostGRA, then we need to rerun the analysis.

Reply:
>If live info has been computed

Does it mean it's been computed in TR_GlobalRegisterAllocator or TR_GlobalLiveVariablesForGC? Or both? 
I found there are cases where the blocks' live locals might not be set in TR_GlobalRegisterAllocator::perform but they're set in TR_GlobalLiveVariablesForGC::perform.

Reply:
The two opts set `cg()->setLiveLocals(liveVars);` to let CG know about the liveness info they computed.